### PR TITLE
Snapping closures

### PR DESCRIPTION
**Pull request by korshaknn**
The Directions API has added a new request parameter that allows routing from/to closed road segments (due to live-traffic closures). The draft documentation of this parameter is [here](https://github.com/mapbox/api-documentation/pull/1078/files) and this is live in production.

We need to support it.

### CODE REVIEW

File: services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/RouteOptionsTest.java [hunk 1]
```diff
@@ -328,6 +328,51 @@ public void annotationsList() {
     assertEquals("congestion,distance,maxspeed,speed", routeOptions.annotations());
   }
 
+  @Test
+  public void snappingClosuresString() {
```
Comment (RingerJK) [inline services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/RouteOptionsTest.java:332]: could you add a test if we set **empty** string and **empty** list via `snappingClosures()`

Comment (korshaknn) [reply services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/RouteOptionsTest.java:333]: done

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/RouteOptions.java [hunk 2]
```diff
@@ -493,6 +493,36 @@ public static Builder builder() {
   @Nullable
   public abstract WalkingOptions walkingOptions();
 
+  /**
+   * A semicolon-separated list of booleans affecting snapping of waypoint locations to road
+   * segments.
+   * If `true`, road segments closed due to live-traffic closures will be considered for snapping.
```
Comment (Guardiola31337) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/RouteOptions.java:499]: Noting that this is Javadocs and \` won't be formatted properly 😞

File: services-directions/src/test/java/com/mapbox/api/directions/v5/MapboxDirectionsTest.java [hunk 3]
```diff
@@ -1414,6 +1415,58 @@ public void withIncidents() throws Exception {
...
+
+    MapboxDirections mapboxDirections = MapboxDirections.builder()
+        .origin(Point.fromLngLat(1.0, 1.0))
+        .addWaypoint(Point.fromLngLat(2.0, 2.0))
+        .destination(Point.fromLngLat(4.0, 4.0))
+        .baseUrl("https://foobar.com")
+        .accessToken(ACCESS_TOKEN)
+        .snappingClosures(snappingClosures)
+        .build();
+    assertNotNull(mapboxDirections);
```
Comment (Guardiola31337) [inline services-directions/src/test/java/com/mapbox/api/directions/v5/MapboxDirectionsTest.java:1433]:
```suggestion

    assertNotNull(mapboxDirections);
```

File: services-directions/src/test/java/com/mapbox/api/directions/v5/MapboxDirectionsTest.java [hunk 4]
```diff
@@ -1414,6 +1415,58 @@ public void withIncidents() throws Exception {
...
+
+    MapboxDirections mapboxDirections = MapboxDirections.builder()
+        .origin(Point.fromLngLat(1.0, 1.0))
+        .addWaypoint(Point.fromLngLat(2.0, 2.0))
+        .destination(Point.fromLngLat(4.0, 4.0))
+        .snappingClosures(snappingClosures)
+        .baseUrl("https://foobar.com")
+        .accessToken(ACCESS_TOKEN)
+        .build();
+    assertNotNull(mapboxDirections);
```
Comment (Guardiola31337) [inline services-directions/src/test/java/com/mapbox/api/directions/v5/MapboxDirectionsTest.java:1450]:
```suggestion

    assertNotNull(mapboxDirections);
```

File: services-directions/src/test/java/com/mapbox/api/directions/v5/MapboxDirectionsTest.java [hunk 5]
```diff
@@ -1414,6 +1415,58 @@ public void withIncidents() throws Exception {
...
+  @Test
+  public void snappingClosuresDefaultValue() throws Exception {
+    MapboxDirections mapboxDirections = MapboxDirections.builder()
+        .origin(Point.fromLngLat(1.0, 1.0))
+        .addWaypoint(Point.fromLngLat(2.0, 2.0))
+        .destination(Point.fromLngLat(4.0, 4.0))
+        .baseUrl("https://foobar.com")
+        .accessToken(ACCESS_TOKEN)
+        .build();
+    assertNotNull(mapboxDirections);
```
Comment (Guardiola31337) [inline services-directions/src/test/java/com/mapbox/api/directions/v5/MapboxDirectionsTest.java:1464]:
```suggestion

    assertNotNull(mapboxDirections);
```

**Review by Guardiola31337 — APPROVED** [review]

**Review by RingerJK — APPROVED** [review]
